### PR TITLE
Add procedures for replacing an SBD disk

### PR DESCRIPTION
**Pull request by tahliar**
### PR creator: Description

Added procedures (disk-based and diskless) for redeploying SBD.  

PDF: 
[cha-ha-storage-protect_en.pdf](https://github.com/user-attachments/files/16022857/cha-ha-storage-protect_en.pdf)
(skip to section 13.11 on page 23)

### PR creator: Are there any relevant issues/feature requests?

* bsc#1216632
* jsc#DOCTEAM-1198

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE-HA 15
  - [x] 15 next *(current `main`, no backport necessary)*
  - [x] 15 SP5
  - [x] 15 SP4
  - [ ] 15 SP3
  - [ ] 15 SP2
- SLE-HA 12
  - [ ] 12 SP5

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Comment by tahliar** [comment]
Render:

image
image
image
image

File: xml/ha_storage_protection.xml [hunk 1]
```diff
@@ -1373,6 +1373,271 @@ Illegal request, Invalid opcode</screen>
...
+    <title>Redeploying disk-based SBD</title>
+    <para>
+      Use this procedure to change diskless SBD to disk-based SBD, to replace an existing
+      SBD device with a new device, or to change the timeout settings for disk-based SBD.
+    </para>
+    <step>
+      <para>
+        Put the cluster into maintenance mode:
+      </para>
+<screen>&prompt.root;<command>crm configure property maintenance-mode=true</command></screen>
```
Comment (zzhou1) [inline xml/ha_storage_protection.xml:1419]: If it is really needed, we can use the simpler interface

`crm maintenance on/off`

**Comment by liangxin1300** [comment]
@zzhou1 @tahliar Setting maintenance seems safer, but stopping cluster service might be unnecessary, the sbd replace process will try to change the CIB like to remove `stonith-sbd` agent when cluster is running, when switching from disk-basd sbd to disk-less sbd

**Comment by zzhou1** [comment]
> @zzhou1 @tahliar Setting maintenance seems safer, but stopping cluster service might be unnecessary, the sbd replace process will try to change the CIB like to remove `stonith-sbd` agent when cluster is running, when switching from disk-basd sbd to disk-less sbd

Somehow, I agree with you @liangxin1300 to use "crm cluster init sbd" stage procedure, which handles more subtle details as you pointed out.

And, `crm cluster start/stop --all` makes sense for the part about "If you need to change any settings" in /etc/sysconfig/sbd.

@tahliar Perhaps we could spin this off into a separate procedure? That fit the section title at the general level.

File: xml/ha_storage_protection.xml (hunk 1 above)
Comment (tahliar) [reply xml/ha_storage_protection.xml:1419]: Thanks @zzhou1! Which versions can use this simpler method? Maintenance mode is mentioned in a few other places, so when this bug is done I'll open another doc task to change the command everywhere else too.

Comment (zzhou1) [reply xml/ha_storage_protection.xml:1419]: Sure. It long exists since SLE12.

**Comment by tahliar** [comment]
> > @zzhou1 @tahliar Setting maintenance seems safer, but stopping cluster service might be unnecessary, the sbd replace process will try to change the CIB like to remove `stonith-sbd` agent when cluster is running, when switching from disk-basd sbd to disk-less sbd
> 
> Somehow, I agree with you @liangxin1300 to use "crm cluster init sbd" stage procedure, which handles more subtle details as you pointed out.
> 
> And, `crm cluster start/stop --all` makes sense for the part about "If you need to change any settings" in /etc/sysconfig/sbd.
> 
> @tahliar Perhaps we could spin this off into a separate procedure? That fit the section title at the general level.

Thanks @liangxin1300 and @zzhou1, good idea! I started out with more procedures, but joined them together because of all the common steps. But if we don't need to stop the cluster services for `crm cluster init sbd`, then it makes sense to split it up into separate procedures again.

File: xml/ha_storage_protection.xml (hunk 1 above)
Comment (tahliar) [reply xml/ha_storage_protection.xml:1419]: Thanks, I'll backport the simpler command all the way to 12 SP5 then :)

**Comment by tahliar** [comment]
@zzhou1 @liangxin1300 Thank you, I've now updated the procedures. It got too long for screen shots so now I've added a PDF to the Description.

File: xml/ha_storage_protection.xml [hunk 2]
```diff
@@ -1373,6 +1373,421 @@ Illegal request, Invalid opcode</screen>
...
+      SBD device with a new device.
+    </para>
+    <step>
+      <para>
+        Put the cluster into maintenance mode:
+      </para>
+<screen>&prompt.root;<command>crm maintenance on</command></screen>
+      <para>
+        In this state, the cluster stops monitoring all resources, so the services managed
+        by the resources can continue to run even if the cluster services stop.
```
Comment (gao-yan) [inline xml/ha_storage_protection.xml:1450]: Would it make sense to point users to the following sections:

https://susedoc.github.io/doc-sleha/main/single-html/SLE-HA-administration/#sec-ha-maint-mode-cluster
https://susedoc.github.io/doc-sleha/main/single-html/SLE-HA-administration/#sec-ha-maint-cluster-stop

of course depending on which suits their case better.

**Review by zzhou1 — APPROVED** [review]
LGTM

File: xml/ha_storage_protection.xml [hunk 3]
```diff
@@ -1373,6 +1373,421 @@ Illegal request, Invalid opcode</screen>
...
+      </para>
+<screen>&prompt.root;<command>crm status</command></screen>
+      <para>
+        Initially, the nodes have a status of <literal>UNCLEAN (offline)</literal>,
+        but after a short time they change to <literal>Online</literal>.
+      </para>
+    </step>
+    <step>
+      <para>
+        Check the SBD configuration. First, check the partition information:
```
Comment (gao-yan) [inline xml/ha_storage_protection.xml:1475]: We could say "check the metadata". The terminology is also used elsewhere BTW.

File: xml/ha_storage_protection.xml [hunk 4]
```diff
@@ -1373,6 +1373,421 @@ Illegal request, Invalid opcode</screen>
...
```
Comment (gao-yan) [inline xml/ha_storage_protection.xml:1582]: Similarly.

File: xml/ha_storage_protection.xml [hunk 5]
```diff
@@ -1373,6 +1373,421 @@ Illegal request, Invalid opcode</screen>
...
+      <para>
+        Stop the cluster services, including SBD, on all nodes:
+      </para>
+<screen>&prompt.root;<command>crm cluster stop --all</command></screen>
+    </step>
+    <step>
+      <para>
+        Reinitialize the device metadata, specifying the new device ID and timeouts as required:
+      </para>
+<screen>&prompt.root;<command>sbd -d /dev/disk/by-id/<replaceable>DEVICE_ID</replaceable> -4 <replaceable>VALUE</replaceable> -1 <replaceable>VALUE</replaceable> create</command></screen>
```
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_storage_protection.xml:1534]: The way this command works is a little awkward - I'm worried that if I write out e.g. `-1 <replaceable>watchdog-timeout</replaceable>`, even though it's in replaceable tags, someone could still read it like watchdog-timeout is the option and -1 is the value. 

You're right that it currently isn't very clear, though, so I could add a little description explaining what they are.

[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_storage_protection.xml]: Thanks @gao-yan, I've made some updates (and rebuilt the PDF).

**Review by gao-yan — APPROVED** [review]
Thanks for the nice revisions, @tahliar 

Only a nitpicking in below.

File: xml/ha_storage_protection.xml [hunk 6]
```diff
@@ -1373,6 +1373,429 @@ Illegal request, Invalid opcode</screen>
...
+      <para>
+        To check the current values, run the following command:
+      </para>
+<screen>&prompt.root;<command>crm configure show</command></screen>
+      <para>
+        If you need to change the values, use the following commands:
+      </para>
+<screen>&prompt.root;<command>crm configure property stonith-watchdog-timeout=0</command>
+&prompt.root;<command>crm configure property stonith-timeout=<replaceable>VALUE</replaceable></command></screen>
+    </step>
```
Comment (gao-yan) [inline xml/ha_storage_protection.xml:1630]: It's occurring to me that for the case of changing diskless SBD to disk-based SBD, we still need to "Configure the SBD STONITH resource" as how it's done in this procedure:

https://susedoc.github.io/doc-sleha/main/single-html/SLE-HA-administration/#pro-ha-storage-protect-fencing

Comment (tahliar) [reply xml/ha_storage_protection.xml:1630]: Uh oh, you're right, thanks for catching that! To keep it simple, I'll just add a step linking to this procedure.

**Pull request merged**